chaco/color_bar.py

```diff
@@ -140,7 +140,10 @@ def _draw_plot(self, gc, view_bounds=None, mode='normal'):
 
             mapper = self.index_mapper
 
-            scrn_points = arange(mapper.low_pos, mapper.high_pos+1)
+            direction = 1 if self.direction == 'normal' else -1
```

Comment:
This does not do the right thing for flipped bounds. For example, for
```python
low_pos = 1
high_pos = -1
```
this gives
```python
array([1])
```

I suggest,
```
low = mapper.low_pos
high = mapper.high_pos
if self.direction == 'flipped':
    low, high = high, low
scrn_points = arange(low, high + 1)
```
and then delete lines 155-156, which would undo the bounds flip.